optimizely/helpers/validator.py

```diff
@@ -185,7 +187,40 @@ def is_attribute_valid(attribute_key, attribute_value):
   if not isinstance(attribute_key, string_types):
     return False
 
-  if isinstance(attribute_value, string_types) or type(attribute_value) in (int, float, bool):
+  if isinstance(attribute_value, (string_types, bool)):
     return True
 
+  if isinstance(attribute_value, (numbers.Integral, float)):
+    return is_finite_number(attribute_value)
+
   return False
+
+
+def is_finite_number(value):
```

Comment:
Are you planning to merge this PR on its own and then update #146 to call `is_finite_number` during audience evaluation? That works for me!

Reply:
Yeah. Once this is approved. I will merge it changes to #146

Reply:
Oops, looks like #146 is merged! But it actually added the calls to `is_finite_number`. So I guess you just need to rebase this branch onto devel in order to add the 2^53/NaN/Infinity limits.